### ISSUE TITLE
freezing in certain DX12 titles with mangohud

### ISSUE THREAD

**Issue by krablord** [flightlessmango/MangoHud#721]
Using latest mangohud on Nvidia 470/latest driver freezes system briefly and then freezes certain DX12 games using different versions of Proton/Wine.
Happens consistently in Hitman 3, Capcom games like: RE Village, Monster Hunter Rise. But never occurs in RE3 and Cyberpunk 2077.
May work fine in menus - and as soon as new assets are being loaded it freezes everything.

Ubuntu 20.04 LTS
Nvidia GPU

Screenshot from moment game is frozen and never recovers - force stopping game causes system to briefly freeze again.
![Screenshot from 2022-03-31 20-33-39](https://user-images.githubusercontent.com/94720169/161115755-cf079933-5061-4cb2-9eeb-cb3332b083be.png)

**Comment by jackun** [comment]
Try reinstalling latest git.

**Comment by krablord** [comment]
did just that - still freezes

**Comment by MCJack123** [comment]
Can confirm this issue is happening in HITMAN 3 for me as well.

`mangohud` 0.6.7.1-4 (AUR build)
Arch Linux latest
NVIDIA GTX 970 on `nvidia` 515.57-7 drivers
HITMAN 3 3.120.0 on Epic Games through Heroic
Proton Experimental 7.0-20220714b

**Comment by Mister-Arkse** [comment]
`mangohud` 0.6.8 (tried latest from git, same problem)
Arch Linux
Kernel 6.2.10-273-tkg-bore
AMD RX Vega 56 on mesa 23.1.0_devel.169578.ba537ac25ad-1
GE-Proton-7-54 (tried latest Proton, same problem)

Game crash every time when the "Health Warning" screen appears when using `gamescope` and `mangohud` together.
Hiding the HUD, the game go beyond the "Health Warning" screen but the graphics are weird and it crashes randomly.

With `mangohud` alone, it goes beyond the "Health Warning" screen but weird graphics and unstable too.

When hiding the HUD, the toggle doesn't work to bring it back.

The "Health Warning" screen occurs after the "compiling shaders" complete.

![2023-04-08_23-04-1680987941](https://user-images.githubusercontent.com/16281589/230742744-31e6c064-aaff-409d-a972-2f8264743cfe.png)
![2023-04-08_23-04-1680987953](https://user-images.githubusercontent.com/16281589/230742751-2730c5d8-0d6f-4306-9111-023d70b321a6.png)

Game works fine if I use `gamescope` and `mangoapp` (`mangoapp` report wrong resolution :raised_eyebrow:)

![2023-04-08_23-04-1680988846](https://user-images.githubusercontent.com/16281589/230743035-7dbcf1a4-0759-47bc-b455-5c005e770bf6.png)

TLDR : 
`gamescope` = works
`mangohud` = weird graphics and unstable
`gamescope` + `mangohud` = crash at "Health Warning" screen
`gamescope` + `mangohud` with HUD hidden = weird graphics and unstable
`gamescope` + `mangoapp` = works

**Comment by flightlessmango** [comment]
The weird graphics is fixed in latest. Check if that also fixes the crashes

**Comment by Mister-Arkse** [comment]
It's working, thanks! I played about 1h of Monster Hunter Rise and no crash.

![2023-04-09_18-04-1681056885](https://user-images.githubusercontent.com/16281589/230784204-3bc7c699-e6e6-4652-8c4c-bc7da314ed08.png)

RE Village Demo works too, played about 30 minutes.

I have to use WINEDLLOVERRIDES='dxgi=b' for those games with `mangohud` otherwise it crashes.

I tried Hitman 3 Demo too (Proton and GE-Proton) but the game doesn't work for me, black screen after launcher with/without `gamescope` and/or `mangohud`
